### PR TITLE
Add README.md and some scripts to the main branch, plus a few changes to cofact.c and Makefile.

### PR DESCRIPTION
**Pull request by gary641**
- Added README.md to the main branch.
- Added link_gwnum script to create links to the gwnum library.
- Added run_all script to run cofact on each Fermat number from F0 to F29 using a reasonably fast method.
   This can be used for regression testing when making code changes.
- In cofact.c change the output message on a successful Suyama test from declaring the cofactor a prime to
    declaring it a probable prime (PRP). Changed the version to 0.8.2.
- In Makefile removed the reference to gwnum.ld since it is no longer needed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+Finally, the cofactor $C$ can be tested to determine if it is a prime power by calculating the greatest common divisor $G = \text{gcd}(A - B, C)$. If $G = 1$ then the cofactor is not a prime power. If $G \neq 1$ then the cofactor is a prime power and is divisible by $G$.
+
+### Computation
+The less performance-critical calculations in cofact use the GNU Multiple Precision (GMP) arithmetic library. However for the heavy lifting of modular squarings required by Pépin's test, the `gwnum` library is used since it is multi-threaded and therefore provides much higher performance. Even so, cofact (main branch) does not have Gerbitz error checking of calculations and does not save interim results to a file. So, for a Fermat cofactor test that is expected to run more than a few hours, it is preferable to first either generate the proof file using mprime / Prime95 (which has Gerbitz error checking and checkpoint / restart capability) or download the proof file from Catherine's [website](https://64ordle.au/fermat/). Once a Fermat number's proof file is in hand, `cofact -upr` can be used to test the new cofactor whenever a new factor of the Fermat number is discovered.
```
Comment (xanthe-cat) [inline README.md:131]: Here, a couple of corrections need to be made to replace Gerbitz with Gerbicz.

In the paragraph below for Proof files – I added $F_{12}$ and $F_{13}$, so we now have the complete set of composite and incompletely factored Fermat numbers up as far as $F_{29}$. (Proof files are probably largely unnecessary below $F_{25}$ but given there’s only so many (18) from $F_{12}$ up to $F_{29}$ it’s probably better to have the whole set available.)

File: README.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+### Proof files
+mprime / Prime95 is capable of generating a verifiable delay function proof file for any Fermat number through $F_{30}$. Proof files have been generated and collected for $F_{14}$ through $F_{29}$ and are available from Catherine’s [website](https://64ordle.au/fermat/). If you wish to use mprime / Prime95 to generate a proof file for $F_{30}$ (requires a system with AVX512 support), we would be [most interested](https://www.mersenneforum.org/showthread.php?t=22668&page=6) in knowing about it!
```
Comment (xanthe-cat) [inline README.md:134]: You might want to change the link to Mersenne forum, which now probably ought to point to: https://www.mersenneforum.org/node/17112/page3

File: README.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+The GMP library is © 1991, 1993–2016, 2018–2024 Free Software Foundation, Inc.
+
+## Mathematics
+A Fermat number is an integer of the form $2^{2^n}+1$, where $n \ge 0$. The five smallest Fermat numbers, $F_0$ through $F_4$, are prime. $F_5$ through $F_{11}$ have been fully factored. $F_{20}$ and $F_{24}$ have been proven composite but not factored, while many other Fermat numbers have known factors.
+
+A Fermat cofactor is the integer $F_n / P$, where $P$ is the product of all known factors of $F_n$.
+
+### Pépin's test
+Pépin's test is a definitive primality test that can be used to determine if a Fermat number is prime. If $F_n$ is the $n^{th}$ Fermat number, then $F_n$ is prime if and only if $3^{(F_n - 1)/2} \equiv -1$ (mod $F$). The term $3^{(F_n - 1)/2}$ (mod $F$) may be calculated by $2^n-1$ squarings of the base $3$, modulo $F_n$. Pépin’s test is also a definitive primality test when performed with bases other than 3, such as 5, 6, and so on.
```
Comment (xanthe-cat) [inline README.md:115]: This paragraph uses Pépin with the acute accent, so if you want to copy / paste this accented version to the rest of the 20 or so uses of Pepin elsewhere in the Readme, you would make my inner pedant very happy!

Also, one of the (mod F_n) lacks the subscript _n, just before "may be calculated by".

Comment (gary641) [reply README.md:115]: Good suggestions. Fixed.

File: README.md (hunk 2 above)
Comment (gary641) [reply README.md:134]: The old link appears to still work for now, but may stop working in the future. So I changed to the new "node" format, but left off page3 which, if the thread grows, would be somewhere in the middle of the thread.

File: README.md (hunk 1 above)
Comment (gary641) [reply README.md:131]: Fixed! Thanks for adding the F12 and F13 proof files.

**Pull request merged**